src/knex.ts

```diff
@@ -1,7 +1,23 @@
-import { knex as Knex } from 'knex'
+import { knex as _knex, Knex } from 'knex'
 import env from './env'
 
-export default Knex({
-	client: 'pg',
-	connection: env.DATABASE_URI
-})
+class SingletonKnex {
+	private static instance: Knex
+	private constructor() {
+		//
+	}
+	public static getInstance(): Knex {
+		if (!env.DATABASE_URI) {
+			return <Knex>{}
```

Comment:
If I throw an Error here it crashes the program on launch. What is your opinion on this @roj1512 ?

Reply:
I don’t know. Why do you call this method on launch?